infer/src/base/CommandLineOption.ml

```diff
@@ -440,6 +443,13 @@ let mk_bool ?(deprecated_no= []) ?(default= false) ?(f= fun b -> b) ?(deprecated
   in
   let default_to_string _ = "" in
   let mk_spec set = Unit (fun () -> set "") in
+  let f =
+    (* also set local config_debug *)
+    if String.equal long "debug" then fun b ->
```

Comment:
I think this logic would better live in `Config`, just because then you don't need to compare to the `"debug"` string: you can rename `config_debug` above to `debug`, export it in CommandLineOption.mli, and set it from `Config` in `debug`'s [callback function](https://github.com/facebook/infer/blob/b71d56715729dcf982c53471fc45373cf59d016d/infer/src/base/Config.ml#L1058-L1060).

The only code duplication then is the default value of `false`.